Doc/Manual/Javascript.html

```diff
@@ -79,7 +79,7 @@ <H3><a name="Javascript_running_swig">28.2.1 Running SWIG</a></H3>
 int gcd(int x, int y);
 extern double Foo;</pre>
 </div>
-<p>To build a Javascript module, run SWIG using the <code>-javascript</code> option and a desired target engine <code>-jsc</code>, <code>-v8</code>, or <code>-node</code>. The generator for <code>node</code> is essentially delegating to the <code>v8</code> generator and adds some necessary preprocessor definitions.</p>
+<p>To build a Javascript module, run SWIG using the <code>-javascript</code> option and a desired target engine <code>-jsc</code>, <code>-v8</code>, <code>-node</code> or <code>-napi</code>. <code>-v8</code> allows for interfacing with a raw embedded version of V8. In this case, it is up to the user to implement a binary module loading protocol. There are two generators supporting Node.js. The older generator for <code>node</code> is essentially delegating to the <code>v8</code> generator and adds some necessary preprocessor definitions. The more recent <code>-napi</code> generator produces <code>node-addon-api</code> that interfaces to Node.js through Node-API. The V8 generator is more mature, while the Node-API generator offers a number of advantages such as binary stable ABI allowing for publishing of universal binary modules on npm, Electron support and automatic multi-threading.</p>
```

Comment:
Reference to "automatic multi-threading" is misleading. NAPI effectively forces modules to be "context-aware," but it doesn't make the generated code MT-safe. More correct formulation would probably be "provides Javascript Workers-aware environment for module developer to utilize."

For reference, MT-safety has multiple meanings in the context.  One is "having multiple module threads interact with the same context," another is "having multiple Javascript Workers interact with the module." NAPI makes the former usable. And in the context of the latter it facilitates per-Worker state maintenance. But both are on the module developer to implement, not on SWIG. In other words there is nothing "automatic" about it, hence the objection. Just in case, it's exactly as "automatic" as #1973. The tests that pass there simply don't have global mutable states, which makes them MT-safe "naturally."

Reply:
Could you post a patch here or a new github issue and I'll get this committed. Thanks.